pytensor/tensor/extra_ops.py

```diff
@@ -751,82 +762,97 @@ def infer_shape(self, fgraph, node, ins_shapes):
                 out_shape = [pt_sum(repeats, dtype=dtype)]
         else:
             if repeats.ndim == 0:
-                out_shape[self.axis] = out_shape[self.axis] * repeats
+                out_shape[axis] = out_shape[axis] * repeats
             else:
-                out_shape[self.axis] = pt_sum(repeats, dtype=dtype)
+                out_shape[axis] = pt_sum(repeats, dtype=dtype)
         return [out_shape]
 
 
-def repeat(x, repeats, axis=None):
-    """Repeat elements of an array.
+def repeat(a: "TensorLike", repeats: TensorLike, axis: int or None) -> TensorVariable:
```

Comment:
Why is one typehint a string but the other is not?

Reply:
No idea, doesn't seem like it needs to be a string

Reply:
Fixed